### DISCUSSION TITLE
Is it possible to adjust the speaking rate?

### DISCUSSION THREAD

**Discussion by godspirit00** [coqui-ai/TTS#756] [General Q&A] [answered]
I wonder if, like many commercial TTS, it is possible to adjust the speaking rate of the generated speech?
I am working on a project in which I need to generate some slower speech. It would help a lot if it is possible to adjust the speaking rate when synthesizing speech.
Thanks!

**Comment by Lizh-TW** [comment]
https://colab.research.google.com/drive/1NC4eQJFvVEqD8L4Rd8CVK25_Z-ypaBHD?usp=sharing#scrollTo=3ZHjyvw9bj52

parper => https://arxiv.org/abs/2005.11129

**Comment by erogol** [comment, marked as answer]
Not for all the models. But for some, you can adjust the speed. `tts` and `tts-server` do not support it yet. You should change the rate in the code or the model config.

**Reply by godspirit00** [reply]
Thanks for the reply! I tried with the GlowTTS model, and it worked. But still, hope that `tts` and `tts-server` can support it one day.

**Comment by seetimee** [comment]
Are there any new ways to speak faster？Thanks

**Comment by pseudotensor** [comment]
What h2oGPT does.  I use pyrubberband.  https://github.com/h2oai/h2ogpt/blob/main/src/tts_utils.py#L74-L105

You don't want to just change the rate, then the sound will not maintain pitch.

**Reply by seetimee** [reply]
thanks,I saw the code.Could you please inform me about the differences between pyrubberband and pydub's speedup? Is pyrubberband considered superior?

**Reply by pseudotensor** [reply]
pydub can only speed-up the sound, not slow it down.

**Reply by seetimee** [reply]
I've already starred this project.haha

**Reply by seetimee** [reply]
gpt3.5 give me an answer that  soundfile.write(audio_speedup_path, y, int(sample_rate * speedup_factor)) also can speedup audio. I'm so confused about what the difference of effects between all the approaches

**Reply by pseudotensor** [reply]
Ya, I went through many packages and attempts.  Speeding up is easier than slowing down, but maintaining pitch etc. is different thing than just speeding up.

**Reply by seetimee** [reply]
I'm trying to speed up audio by 2x.But most approaches just speeding up the audio without preserving pitch.I believe it still requires a model-based solution.

**Reply by pseudotensor** [reply]
pyrubberband does very well even for speeding up.  But please try and judge for yourself.

**Reply by seetimee** [reply]
Thanks a lot.

**Reply by seetimee** [reply]
pyrubberband is much better than pydub and moviepy. Thanks bro.
